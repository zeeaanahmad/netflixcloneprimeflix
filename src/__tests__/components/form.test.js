import React from "react";
import { render } from "@testing-library/react";
import { Form } from "../../components";

jest.mock("react-router-dom", () => ({
  Link: "Link",
  Route: ({ children, path }) => children({ match: path === "/signup" }),
}));

describe("<Form />", () => {
  it("renders the <Form /> with populated data", () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Form>
        <Form.Title>Sign In Now</Form.Title>
        <Form.Base>
          <Form.Input placeholder="Email address" />
          <Form.Input type="password" placeholder="Password" />
          <Form.Submit disabled type="submit">
            Sign In
          </Form.Submit>
        </Form.Base>
        <Form.Text>
          New to Netflix? <Form.Link to="/signup">Sign up now.</Form.Link>
        </Form.Text>
        <Form.TextSmall>
          This page is protected by Google reCAPTCHA to ensure you're not a bot.
          Learn more.
        </Form.TextSmall>
      </Form>
    );

    expect(
      getByText(
        "This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more."
      )
    ).toBeTruthy();
    expect(getByText("Sign In Now")).toBeTruthy();
    expect(getByText("Sign In")).toBeTruthy();
    expect(getByText("Sign In").disabled).toBeTruthy();
    expect(getByPlaceholderText("Email address")).toBeTruthy();
    expect(getByPlaceholderText("Password")).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Form/> with an error", () => {
    const { container, getByText, queryByText } = render(
      <Form>
        <Form.Error>Your email address is wrong.</Form.Error>
        <Form.Submit type="submit">Sign In</Form.Submit>
      </Form>
    );

    expect(getByText("Your email address is wrong.")).toBeTruthy();
    expect(queryByText("Sign In").disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
