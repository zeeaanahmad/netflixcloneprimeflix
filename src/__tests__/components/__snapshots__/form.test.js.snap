// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders the <Form /> with populated data 1`] = `
<div
  class="form__Container-sc-1gkudw1-0 cGfOGK"
>
  <h1
    class="form__Title-sc-1gkudw1-3 cYJqjk"
  >
    Sign In Now
  </h1>
  <form
    class="form__Base-sc-1gkudw1-1 blcOFQ"
  >
    <input
      class="form__Input-sc-1gkudw1-7 dChWbe"
      placeholder="Email address"
    />
    <input
      class="form__Input-sc-1gkudw1-7 dChWbe"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-1gkudw1-8 hDfNkC"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-sc-1gkudw1-4 gAbsDR"
  >
    New to Netflix? 
    <link
      class="form__Link-sc-1gkudw1-6 gLjvGn"
      to="/signup"
    >
      Sign up now.
    </link>
  </p>
  <p
    class="form__TextSmall-sc-1gkudw1-5 iiFVRZ"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
  </p>
</div>
`;

exports[`<Form /> renders the <Form/> with an error 1`] = `
<div
  class="form__Container-sc-1gkudw1-0 cGfOGK"
>
  <div
    class="form__Error-sc-1gkudw1-2 CnAHo"
  >
    Your email address is wrong.
  </div>
  <button
    class="form__Submit-sc-1gkudw1-8 hDfNkC"
    type="submit"
  >
    Sign In
  </button>
</div>
`;
